src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+            }
+            is SuperstructureRequest.SingleSubstationIntake -> {
+              when (usingGamePiece) {
+                GamePiece.CONE -> SuperstructureStates.SINGLE_SUBSTATION_INTAKE_CONE
+                GamePiece.CUBE -> SuperstructureStates.SINGLE_SUBSTATION_INTAKE_CUBE
+                GamePiece.NONE -> SuperstructureStates.IDLE
+              }
+            }
+            is SuperstructureRequest.Tuning -> SuperstructureStates.TUNING
+            else -> currentState
+          }
+      }
+      SuperstructureStates.HOME_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.stowedDownAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition) {
+          nextState = SuperstructureStates.HOME
+        }
+      }
+      SuperstructureStates.HOME -> {
+        // Outputs
+        elevator.currentRequest = Request.ElevatorRequest.Home()
+        manipulator.currentRequest = Request.ManipulatorRequest.Home()
+
+        // Transition
+        if (elevator.isHomed && manipulator.isHomed) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CUBE_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.intakeAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.groundIntakeCubeExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CUBE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CUBE -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.intakeAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.intakeVoltage.get()
+          )
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.groundIntakeCubeExtension.get(),
+            Manipulator.TunableManipulatorStates.cubeInVoltage.get()
+          )
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
```

Comment:
we should not switch to cleanup unless the request changes or we think we have gotten the game piece, right? Otherwise we will immediately transition and never actually intake